main/src/ca/uwaterloo/flix/language/ast/TypeConstructor.scala

```diff
@@ -177,8 +181,19 @@ object TypeConstructor {
   /**
     * A type constructor that represent the type of enums.
     */
+  @IntroducedBy(Kinder.getClass)
   case class Enum(sym: Symbol.EnumSym, kind: Kind) extends TypeConstructor
 
+  /**
+    * An unkinded type constructor that represent the type of enums.
+    */
+  @EliminatedBy(Kinder.getClass)
+  case class UnkindedEnum(sym: Symbol.EnumSym) extends TypeConstructor {
+
+    @deprecated("An UnkindedEnum has no associated kind.")
```

Comment:
Why deprecated? I can understand its an error.. but why deprecated?

Reply:
It's a slightly stronger compile-time guarantee. If there's somewhere that we know statically is an UnkindedEnum and we call `kind` on it, we'll get a compile-time warning/error rather than runtime. Inspired by Google Guava's [deprecating unsupported operations](https://guava.dev/releases/21.0/api/docs/com/google/common/collect/ImmutableList.html#add-int-E-) on `ImmutableList`s.

Reply:
Ok

Reply:
But that's abuse though.

Reply:
I'll remove it.